### PR TITLE
Added adminme command.

### LINKED ISSUE #26: Give admin permissions in bridged room
In every bridged chat, I noticed, the discordpuppet is an admin, while I am merely default power level. I find this strangely restrictive. Is there a reason why we don't want to allow the user to mess with the room's settings and members?

### PR DESCRIPTION
**Pull request by dylhack**
Fixes matrix-discord/mx-puppet-discord#26

### CODE REVIEW

**Comment by dylhack** [comment]
Note: This is missing room alias support.

**Review by Sorunome — CHANGES_REQUESTED** [review]
Thank you for the PR! There are, however, a few things:

1. Please put the stuff that actually does the admin setting, based on simple parameters, into the provisioner. That way other classes can easily re-use that code. You can make it e.g. reply true on success and false on failure, allowing you to output messages.
2. if you use `withPid: false` you can then use `this.bridge.roomSync.resolve` to turn the entire params string into an `IRemoteRoom` - you will then have to check the existance of said room, if you own the puppet AND if you are in the room, then. That will automatically make aliases and so much more work, though.

File: src/botprovisioner.ts [hunk 1]
```diff
@@ -321,6 +321,42 @@ Usage: \`list\``,
...
+				// Check to see if this user is in the room before providing a power level.
+				const members = await client.getJoinedRoomMembers(param);
+				if (!members.includes(userId)) {
+					await sendMessage("You're not in this room.");
+					return;
+				}
+
+				// Then finally set power level 100 (admin)
+				// tslint:disable-next-line:no-magic-numbers
+				await client.setUserPowerLevel(userId, param, 100);
```
Comment (Sorunome) [inline src/botprovisioner.ts]: No need to disable tslint for that line!
```ts
const POWER_LEVEL_OP = 100;
await client.setUserPowerLevel(userId, param, POWER_LEVEL_OP);
```

File: src/botprovisioner.ts [hunk 2]
```diff
@@ -321,6 +321,42 @@ Usage: \`list\``,
...
+				// Puppet to refer to.
+				const puppet = await this.provisioner.get(puppetId);
+				if (!puppet) {
+					await sendMessage("Puppet not found.");
+					return;
+				}
+				// Owner of the puppet (The user to make admin).
+				const userId = puppet.puppetMxid;
+				// This Matrix client will allow us to make the user admin.
+				const client = await this.bridge.roomSync.getRoomOp(param);
```
Comment (Sorunome) [inline src/botprovisioner.ts]: ` || this.bridge.botIntent.underlyingClient`

Comment (Sorunome) [reply src/botprovisioner.ts]: Also, `getRoomOp` actually takes internal MXIDs (the ones starting with `!`)

File: src/botprovisioner.ts [hunk 3]
```diff
@@ -323,31 +323,9 @@ Usage: \`setmatrixtoken <token>\``,
...
-					return;
-				}
-				// Check to see if this user is in the room before providing a power level.
-				const members = await client.getJoinedRoomMembers(param);
-				if (!members.includes(userId)) {
-					await sendMessage("You're not in this room.");
-					return;
-				}
+				// Set the user to admin
+				await this.provisioner.setAdmin(puppetId, param);
```
Comment (Sorunome) [inline src/botprovisioner.ts]: How about use the returned boolean to display a message if setting as admin failed? ^^

Comment (dylhack) [reply src/botprovisioner.ts]: Resolved [here](https://github.com/dhghf/mx-puppet-bridge/commit/71dea5abe8cdcf6a93c8cdbf2247a2f838faab51)

**Comment by Sorunome** [comment]
Thanks a lot!

**Pull request merged**